Update basic template to use the latest Melange version (OCaml 4.14)

laaksonel opened the pull request:
Generated dune.ml ended up in the following error:
`File "dune.mel", line 352, characters 80-98:
352 |  (mode (promote (until-clean) (only client.bs.js)))(deps (:inputs  client.ast)  (include client.d) ../bsconfig.json ../.melange.eobjs/MelangeBasicProject.cmi(alias ../node_modules/reason-react/src/bsb_world)(alias ../node_modules/reason-react/src/legacy/bsb_world))`

So I concluded dune is supposed to be 3.1

Reason dependency is updated to the same version what melange uses in order to avoid conflicting versions.

laaksonel committed:
Updated dune from 2.8 to 3.1

laaksonel committed:
Update dependencies

laaksonel committed:
Update reason version to 3.8.0

Use the same version as melange in order to avoid the following
conflict:

Conflicting constraints:
  melange-project -> @opam/reason@github:reasonml/reason#3.8.0
  melange-project -> melange -> @opam/reason@github:reasonml/reason:reason.opam#3.8.0

laaksonel committed:
Fix melange OCaml conflict

anmonteiro commented:
Let's use melange from the tip, which contains more fixes, and runs on OCaml 4.14

laaksonel committed:
Update to current Melange tip commit

- Update OCaml to 4.14
- Resolve Reason dependency to the current tip commit of reasonml/reason
  Otherwise build fails to resolve Reason_omp module.

laaksonel commented:
Sure, I was a bit hesitant to directly use main branch. I didn't have courage to assume that Melange main/fork branch is always stable. Is it safe to assume so?

Alternatively I'd suggest that there should be at least a tag to indicate the latest stable commit.

anmonteiro commented:
Doing this in #9 because it's easier for me to test.
